Clover hpc

sheridanfew opened the pull request.

sheridanfew committed:
Clean up inconsistency in type of inverted commas. This has caused problems with some compilers.
Clean up anomylous naming and spacing. This is in order to allow automatic editing of scripts matching given patterns in later changes.

Used the following commands to achieve this in shell terminal in CLOVER directory:

# Clean up inconsistency in type of inverted commas. This has caused problems with some compilers.

SRC="‘"
DST="'"
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py
SRC="’"
DST="'"
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

# Clean up anomylous naming and spacing. This is in order to allow automatic editing of scripts matching given patterns in later changes.

sed -i "" -e "s/location_input_data/location_inputs/g" Scripts/*/*py
sed -i "" -e "s/_inputs =  pd.read_csv/_inputs = pd.read_csv/g" Scripts/*/*py
sed -i "" -e "s/_inputs  = pd.read_csv/_inputs = pd.read_csv/g" Scripts/*/*py

sheridanfew committed:
no message

sheridanfew committed:
Reconfigure to take '.' as filepath

Reconfigure to take '.' as filepath - necessary for HPC where CLOVER is copied to the node before processing. (May wish to amend if CLOVER dir gets very big)

Commands used:

SRC="\/\*\*\*YOUR LOCAL FILE PATH\*\*\*\/CLOVER 4.0"
DST="."
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

SRC="\/\*\*\*YOUR LOCAL FILE PATH\*\*\*\/CLOVER"
DST="."
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

# Reconfigure to take kwargs for location
sed -i "" -e 's/__init__(self)/__init__(self,**kwargs)/g' Scripts/*/*py

SRC="self.location = 'Bahraich'"
DST="self.location = kwargs.get('location')"
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

SRC="self.location = ‘Bahraich’" # This is required too as there is some inconsistency in type of inverted commas used in original CLOVER scripts.
DST="self.location = kwargs.get('location')"
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

sheridanfew committed:
Revert "Reconfigure to take '.' as filepath"

This reverts commit 599cc854d2019298610403d7690e07b3eb97d9d0.

sheridanfew committed:
Reconfigure to take '.' as filepath - necessary for HPC

# Reconfigure to take '.' as filepath - necessary for HPC where CLOVER is copied to the node before processing. (May wish to amend if CLOVER dir gets very big)

Bash commands:

SRC="\/\*\*\*YOUR LOCAL FILE PATH\*\*\*\/CLOVER 4.0"
DST="."
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

SRC="\/\*\*\*YOUR LOCAL FILE PATH\*\*\*\/CLOVER"
DST="."
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

sheridanfew committed:
Reconfigure to take kwargs for location

# Reconfigure to take kwargs for location
sed -i "" -e 's/__init__(self)/__init__(self,**kwargs)/g' Scripts/*/*py

SRC="self.location = 'Bahraich'"
DST="self.location = kwargs.get('location')"
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

SRC="self.location = ‘Bahraich’" # This was required too as there was some inconsistency in type of inverted commas used in original CLOVER scripts - may have been fixed by earlier commands
DST="self.location = kwargs.get('location')"
sed -i "" -e "s/$SRC/$DST/g" Scripts/*/*py

# Set up dependencies in taking kwargs for location

sed -i "" -e "s/Diesel()\./Diesel(kwargs)./g" Scripts/*/*py
sed -i "" -e "s/Grid()\./Grid(kwargs)./g" Scripts/*/*py
sed -i "" -e "s/Solar()\./Solar(kwargs)./g" Scripts/*/*py
sed -i "" -e "s/Finance()\./Finance(kwargs)./g" Scripts/*/*py
sed -i "" -e "s/GHGs()\./GHGs(kwargs)./g" Scripts/*/*py
sed -i "" -e "s/Load()\./Load(kwargs)./g" Scripts/*/*py
sed -i "" -e "s/Energy_System()\./Energy_System(kwargs)./g" Scripts/*/*py

sheridanfew committed:
Replace spaces with underscores in filenames

Replace spaces with underscores in filenames because spaces cause problems in file handling

This stops github from recognising that files are the same, which is a pity! This will need to be resolved to merge hpc CLOVER with new features in main branch

Bash commands:

rename -s ' scripts' '_scripts' Scripts/*/

for file in Scripts/*/*py;
do
   sed "s| scripts|_scripts|" $file > temp; mv temp $file;
done

sheridanfew committed:
Set up scripts to allow specification of inputs with kwargs (overrides those specified in location)

# Set up scripts to allow specification of inputs with kwargs (overrides those specified in location)

bash commands used:

for name in grid location finance GHG device optimisation energy_system scenario diesel
do
	cat > ${name}_line.py << EOF
        # Replace input values with keywords if specified
        if kwargs.get('${name}_inputs'):
            for i in kwargs.get('${name}_inputs'):
                if not i[0] in self.${name}_inputs.index:
                    print("Couldn't find entry",i[0],"in ${name}_inputs. Perhaps it's misspelt in kwargs? Printing list of possible variables and exiting.")
                    print(self.${name}_inputs.index)
                    exit(1)
            self.${name}_inputs[i[0]] = i[1]
EOF

	for file in Scripts/*/*py
	do
		sed "/        self.${name}_inputs = pd.read_csv/r ${name}_line.py" $file > tmp
		mv tmp $file
	done

rm ${name}_line.py

done

sheridanfew committed:
Add bash file used to generate CLOVER-hpc from (old) version of CLOVER

sheridanfew committed:
Miscellanious other changes

Relating to specific projects - I think unlikely to be harmful, but may not be useful

sheridanfew committed:
Put in CLOVER-master dir

sheridanfew committed:
Put example jobs in place

Bhinjpur

sheridanfew committed:
Update README.md

sheridanfew committed:
Update README.md

sheridanfew committed:
Update Location inputs.csv

sheridanfew committed:
Update README.md

sheridanfew committed:
Rearranging files & revising launch scripts

sheridanfew committed:
Delete Bhinjpur_LiB_cyc1000_cost1270_It2yrs_Opt.py

sheridanfew committed:
Update launch_CLOVER_job_multi_share.sh

Accomodate revised file structuring (keep Jobs in separate dir and repository from main CLOVER files

sheridanfew committed:
Remove Jobs (added to separate repo)

sheridanfew committed:
Update to launch scripts & readme, inclusion of simple run_info analysis script

sheridanfew committed:
Update Devices.csv

Reduced number of devices for example run

sheridanfew committed:
Update README.md

sheridanfew committed:
Update README.md

sheridanfew committed:
Update README.md

sheridanfew committed:
Update README.md

sheridanfew committed:
Update README.md

sheridanfew committed:
Update README.md

sheridanfew committed:
Update launch_CLOVER_job_multi_share.sh

sheridanfew committed:
Corrected launch script

sheridanfew committed:
Update launch_CLOVER_job_multi_share.sh

Clean up and improve comments

sheridanfew committed:
Update README.md

sheridanfew committed:
Update HPC_setup.sh

sheridanfew committed:
Merge branch 'CLOVER-hpc' of https://github.com/sheridanfew/CLOVER into CLOVER-hpc

sheridanfew committed:
Change instructions & launch script from 'work' to 'home' dir

New hpc users don't seem to have a work directory of the same format

sheridanfew committed:
Corrections to path for home directory

Directory structure changed in 2018

sheridanfew committed:
Update HPC_setup.sh

sheridanfew committed:
Update launch_CLOVER_job_multi_share.sh

sheridanfew committed:
Update HPC_setup.sh

Changed file format to remove \r newlines which were causing problems

sheridanfew committed:
Changes following training session

- Moved launch file & added it's new home to the path in .bash_profile (readme instructions, could also add to setup file)
- Corrected time between renewables ninja requests in Load_Solar_Bhinjpur.py

sheridanfew committed:
Update README.md

sheridanfew committed:
Update Load_Solar_Bhinjpur.py

sheridanfew committed:
Remove old mess

sheridanfew committed:
Create folder Saved optimisations

sheridanfew committed:
Add placeholder file in saved opt & sim

Hoping this will encourage these necessary folders to be recognised in upload & download of repo

sheridanfew committed:
Make directory when saving opt/sim if they don't exist

sheridanfew committed:
Update HPC_setup.sh

sheridanfew committed:
Revert underscores to spaces to allow comparison with main clover branch

BenWinchester commented on Load_Solar_Bhinjpur.py:
```diff
@@ -0,0 +1,36 @@
+import time
+import os
+
+# Go to CLOVER directory
```
Change this :(

BenWinchester commented on Scripts/Generation_scripts/Diesel.py:
```diff
@@ -18,15 +18,23 @@
 
 class Diesel():
-    def __init__(self): 
+    def __init__(self,**kwargs): 
```
```
def __init__(self, *, location, diesel_inputs=None):
   self.location = location
```

BenWinchester replied:
:)

BenWinchester commented on Scripts/Generation_scripts/Diesel.py:
```diff
@@ -18,15 +18,23 @@
         self.diesel_inputs = pd.read_csv(self.diesel_filepath,header=None,index_col=0).round(decimals=3)
-
+        # Replace input values with keywords if specified
+        if kwargs.get('diesel_inputs'):
```
```
if diesel_input is not None
```

BenWinchester replied:
If the kwarg is specified as above, then this would default to `None` if the kwarg was not specified.

BenWinchester commented on Scripts/Generation_scripts/Diesel.py:
```diff
@@ -18,15 +18,23 @@
-        self.location = 'Bahraich'
-        self.CLOVER_filepath = '/***YOUR LOCAL FILE PATH***/CLOVER 4.0'
+        self.location = kwargs.get('location')
+        self.CLOVER_filepath = '.'
```
```
os.getcwd()
```

BenWinchester replied:
Python's `os` module contains built-in functionality for getting the working directory.

sheridanfew committed:
Clean Up

Added indication of which kwargs can/must be specified per script (some way towards Ben's suggested safer way of specifying arguments)

Removed some of my obsolete options (quick battery recharge, efficient diesel - introduced in the earlier "miscellaneous" commit):

Left a few other additions in (which were also introduced in the earlier "miscellaneous" commit) :
- In Optimisation.py: Additional metrics around storage performance reported: LCSE, Cumulative storage cost
- In Load.py: optional arguments for different approaches to load generation
- In Optimisation.py: optional argument to dump spare battery capacity at the end of a simulation period

sheridanfew committed:
Merge branch 'master' into CLOVER-hpc_to_merge

sheridanfew committed:
Re-add Bahraich location, update readme, rename launch script, remove redundant script

sheridanfew committed:
Revert " " to "_" in paths to scripts

sheridanfew committed:
Update UPDATE REGISTER.rtf

sheridanfew committed:
Update README.md

sheridanfew committed:
Merge pull request #1 from sheridanfew/CLOVER-hpc_to_merge

Clover hpc to merge

sheridanfew committed:
Update Energy_System.py

sheridanfew committed:
Yearly load stats as variable input
